### ISSUE TITLE
Exceptions are obscure when using `$ref` inside `anyOf`, `oneOf`, `allOf`, etc...

### ISSUE THREAD

**Issue by abravalheri** [horejsek/python-fastjsonschema#141]
Hello, when validating composite schemas with `fastjsonschema` (i.g. `oneOf`, `allOf`, etc..) the error messages can be very tricky to understand. In those scenarios I always try to reconstruct some meaningful error messages by looking at `exception_obj.definition`.

However, that is very hard to do when using `$ref` to organise the schemas in smaller pieces, specially if those smaller pieces are stored in separated schemas that get re-used.
This happens because both the error message and the exception attribute will only briefly mention the value of `$ref` itself, instead of the actual schema that is not being matched...

This behaviour is shown in the example bellow:

```pycon
>>> from pprint import pprint
>>>
>>> import fastjsonschema
>>>
>>> repo = {
...     "sch://m3": {"type": "number", "multipleOf": 3},
...     "sch://m5": {"type": "number", "multipleOf": 5},
... }
>>> schema = {
...     "$schema": "http://json-schema.org/draft-07/schema",
...     "anyOf": [
...         {"$ref": "sch://m3"},
...         {"$ref": "sch://m5"},
...     ]
... }
>>>
>>> fastjsonschema.validate(schema, 2, handlers={"sch": repo.__getitem__})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "~/workspace/python-fastjsonschema/fastjsonschema/__init__.py", line 114, in validate
    return compile(definition, handlers, formats, use_default)(data)
  File "<string>", line 16, in validate
fastjsonschema.exceptions.JsonSchemaValueException: data must be valid by one of anyOf definition
>>>
>>> try:
...    fastjsonschema.validate(schema, 2, handlers={"sch": repo.__getitem__})
... except fastjsonschema.JsonSchemaValueException as ex:
...     print(ex)
...     print("Rule:", ex.rule)
...     print("Definition:")
...     pprint(ex.definition)
...
data must be valid by one of anyOf definition
Rule: anyOf
Definition:
{'$schema': 'http://json-schema.org/draft-07/schema',
 'anyOf': [{'$ref': 'sch://m3'}, {'$ref': 'sch://m5'}]}
>>>
```

Please note that the user seeing the error, or the API consumer needs to have access to all the JSONSchema documents to understand what is going wrong...

We can compare this behaviour with the other popular library:

```pycon
>>> import jsonschema
>>> resolver = jsonschema.RefResolver.from_schema({}, handlers={"sch": repo.__getitem__})
>>> jsonschema.validate(2, schema=schema, resolver=resolver)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "~/workspace/python-fastjsonschema/.venv/lib/python3.8/site-packages/jsonschema/validators.py", line 1022, in validate
    raise error
jsonschema.exceptions.ValidationError: 2 is not a multiple of 3

Failed validating 'multipleOf' in schema[0]:
    {'multipleOf': 3, 'type': 'number'}

On instance:
    2
>>>
```

Although the error here is more imprecise than the one provided by `fastjsonschema`, it is also more clear because it prints the offending rule, expanding `$ref`.

### TLDR
If `fastjsonschema` expand `$ref` in the `exception_obj.definitions`, it would be much easier to understand, deal with value exceptions...
